cuda_core/tests/test_module.py

```diff
@@ -245,3 +246,14 @@ def test_num_args_error_handling(deinit_all_contexts_function, cuda12_prerequisi
     with pytest.raises(CUDAError):
         # assignment resolves linter error "B018: useless expression"
         _ = krn.num_arguments
+
+
+def test_module_serialization_roundtrip(get_saxpy_kernel):
+    _, objcode = get_saxpy_kernel
+    result = pickle.loads(pickle.dumps(objcode))  # nosec B403, B301
+
+    assert isinstance(result, ObjectCode)
+    assert result._handle is None
+    assert objcode.code == result.code
+    assert objcode._sym_map == result._sym_map
+    assert objcode._code_type == result._code_type
```

Comment:
Should these have a public property that exposes them? They are user inputs but I don't see a non-private way of getting them back out.

Reply:
> Should these have a public property that exposes them?

I think "No": It's OK to view serialization/deserialization as an implementation detail.

Reply:
I think outside of the context of serialization, it would be good to allow a user to introspect what type of code is being represented by the `ObjectCode` instance. But for the sake of this PR we don't need to address this.